File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        stack = []
        ret = [0] * len(T)
        
        for i in range(len(T)):
            while stack and T[i] > T[stack[-1]]:
                curr = stack.pop()
                ret[curr] = i - curr
            stack.append(i)
            
        return ret

File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> res(n);
        stack<int> st;
        for (int i = 0; i < n; i++) {
            int temp = -1;
            if(!st.empty()) temp = st.top();
            while (!st.empty() && temperatures[temp] < temperatures[i]) {
                res[st.top()] = i - st.top();
                st.pop();
                if(!st.empty()) temp = st.top();
            }
            st.push(i);
        }
        return res;
    }
};